stat/spatial/spatial.go

```diff
@@ -43,11 +42,21 @@ func GetisOrdGStar(i int, data, weights []float64, locality mat.Matrix) float64
 	n := float64(len(data))
 	mean, std := stat.MeanStdDev(data, weights)
 	var dwd, dww, sw float64
-	for j, v := range data {
-		w := locality.At(i, j)
-		sw += w
-		dwd += w * v
-		dww += w * w
+	if band, ok := locality.(mat.Banded); ok {
```

Comment:
Yes, I have thought that that might be useful. However, I don't see a way to define an arbitrary set of non-zero elements that reduces from O(n^2) time (space can be trivially - a type wrapping `*mat.BandDense` for example that is not a `mat.Banded` and uses those `*` elements).

We should make an issue to design/implement the function you're suggesting, but I don't think it will capture all cases without being entirely ad hoc, or unwieldy for the user to implement the interface.